### ISSUE TITLE
Improve output with WARN_AS_ERROR = FAIL_ON_WARNINGS

### ISSUE THREAD

**Issue by rburchell** [doxygen/doxygen#8747]
**Describe the bug**
Given something that produces a warning, like `\abrief foo` (no such command 'abrief'), I would expect that `WARN_AS_ERROR = FAIL_ON_WARNINGS` should give me output that reflected that failure (i.e. turn them into errors).

Instead, I get this:
`foo.h:16: warning: Found unknown command '\abrief'`

**Expected behavior**
`WARN_AS_ERROR = yes` produces this, which I think is better:

`foo.h:16: error: Found unknown command '\abrief' (warning treated as error, aborting now)`

**Version**
Doxygen 1.9.1 on Fedora 34, distribution provided (but I have a hard time believing that the platform is relevant).

**Comment by albert-github** [comment]
The idea behind `WARN_AS_ERROR = FAIL_ON_WARNINGS` is that doxygen continues but at the end sets the exit status to a non zero status in case of a detected warning, contrary to when using `WARN_AS_ERROR = NO` where doxygen continues as well but leaves the exit status at 0.
In case of `WARN_AS_ERROR = YES` doxygen is directly terminated with a non zero exit status.

A message like `(warning treated as error, aborting now)` would not be correct as the process is not terminated directly.

When we look at the output of the different situations we see:

**`WARN_AS_ERROR = FAIL_ON_WARNINGS`**
```
...
Generating code for file aa.h...
Generating file documentation...
Generating docs for file aa.h...
D:/speeltuin/issue_8747/aa.h:3: warning: Found unknown command '\abrief'
Generating page documentation...
...
lookup cache used 0/65536 hits=0 misses=0
finished...
Exiting...
```

**`WARN_AS_ERROR = NO`**
```
...
Generating code for file aa.h...
Generating file documentation...
Generating docs for file aa.h...
D:/speeltuin/issue_8747/aa.h:3: warning: Found unknown command '\abrief'
Generating page documentation...
...
lookup cache used 0/65536 hits=0 misses=0
finished...
```

**`WARN_AS_ERROR = YES`**
```
...
Generating code for file aa.h...
Generating file documentation...
Generating docs for file aa.h...
D:/speeltuin/issue_8747/aa.h:3: error: Found unknown command '\abrief' (warning treated as error, aborting now)
Exiting...
```

So I think there is also a distinction through the `Exiting...`.

**Comment by rburchell** [comment]
> A message like "warning treated as error, aborting now" would not be correct as the process is not terminated directly.

I think you're reading my message a little too literally, so let me attempt to clarify.

The part that I directly have an issue with is specifically the "warning:" prefix, not the "aborting now" part. When doxygen is treating a warning as an error, it should call it an error, as it makes identifying the exact cause of a failure easier. If that is done, one does not need to know that some of doxygen's warnings are actually errors. This is important for end users, but even more important when running in the context of an IDE that may group errors and warnings differently, for example.

In my mind, FAIL_ON_WARNINGS is basically equivilent to a compiler's -Werror (see below).

> So I think there is also a distinction through the `Exiting...`.

Yes, you are right: but I don't think that everything responsible for running Doxygen should have to work so hard to figure out what warnings are being treated as errors.

---

Regarding `-Werror`: compare doxygen's behaviour with that of a compiler, for example:

```
bluebox:~/% cat test.c
int main(int argc, char **argv)
{
    foo();

    bar();
}

bluebox:~/% gcc -o test test.c
test.c: In function ‘main’:
test.c:3:5: warning: implicit declaration of function ‘foo’ [-Wimplicit-function-declaration]
    3 |     foo();
      |     ^~~
test.c:5:5: warning: implicit declaration of function ‘bar’ [-Wimplicit-function-declaration]
    5 |     bar();
      |     ^~~
/bin/ld: /tmp/cc9ELIpr.o: in function `main':
test.c:(.text+0x15): undefined reference to `foo'
/bin/ld: test.c:(.text+0x1f): undefined reference to `bar'
collect2: error: ld returned 1 exit status
zsh: exit 1                                                                                                            

bluebox:~/% gcc -o test test.c -Werror
test.c: In function ‘main’:
test.c:3:5: error: implicit declaration of function ‘foo’ [-Werror=implicit-function-declaration]
    3 |     foo();
      |     ^~~
test.c:5:5: error: implicit declaration of function ‘bar’ [-Werror=implicit-function-declaration]
    5 |     bar();
      |     ^~~
cc1: all warnings being treated as errors
zsh: exit 1                                                                                                            
```

**Comment by albert-github** [comment]
So currently we have:

**`WARN_AS_ERROR = FAIL_ON_WARNINGS`**
```
.../aa.h:3: warning: Found unknown command '\abrief'
```

**`WARN_AS_ERROR = NO`**
```
.../aa.h:3: warning: Found unknown command '\abrief'
```

**`WARN_AS_ERROR = YES`**
```
.../aa.h:3: error: Found unknown command '\abrief' (warning treated as error, aborting now)
```


So you would like to have:
**`WARN_AS_ERROR = FAIL_ON_WARNINGS`**
```
.../aa.h:3: error: Found unknown command '\abrief'
```

where the word `warning` is replaced by `error` for `WARN_AS_ERROR = FAIL_ON_WARNINGS` like it is also done for ``WARN_AS_ERROR = YES`

**Comment by rburchell** [comment]
Yes, exactly.

It could also be:
 
`WARN_AS_ERROR = FAIL_ON_WARNINGS`

``` 
.../aa.h:3: error: Found unknown command '\abrief' (warning treated as error)
```

To give a hint about the WARN_AS_ERROR option, but I personally don't find the bracketed part valuable.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8748

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
